torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,251 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Landsat 7 Cloud Cover Assessment Validation Data"""
+import abc
+import glob
+import hashlib
+import os
+from functools import lru_cache
+from typing import Any, Callable, Dict, List, Optional, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.colors import ListedColormap
+from PIL import Image
+from rasterio.crs import CRS
+from torch import Tensor
+from torch.utils.data import Dataset
+
+from .geo import NonGeoDataset, RasterDataset
+from .landsat import Landsat7
+from .utils import BoundingBox, download_url, extract_archive, working_dir
+
+
+class L7Irish(RasterDataset):    
+    """L7 Irish dataset.
+
+    The `L7 Irish <https://landsat.usgs.gov/landsat-7-cloud-cover-assessment-validation-data>`__ dataset is based on Landsat 7
+
+    Dataset format:
+    * # images: 
+    * image size: 
+    * # spectral bands: 
+    * Level-1G
+    
+    * labels are values [0, 64, 128, 192, 255]
+
+     Dataset classes:
+     * Fill
+     * Cloud Shadow
+     * Clear
+     * Thin Cloud
+     * Cloud
+
+     If you use this dataset in your research, please cite the following papers:
+
+     * https://doi.org/10.5066/F7XD0ZWC
+     * https://doi.org/10.1109/TGRS.2011.2164087
+     """
+    
+    # naming... [need to change]
+    filename_regex = r"""
+        ^L
+        (?P<sensor>[COTEM])
+        (?P<satellite>\d{2})
+        _(?P<processing_correction_level>[A-Z0-9]{4})
+        _(?P<wrs_path>\d{3})
+        (?P<wrs_row>\d{3})
+        _(?P<date>\d{8})
+        _(?P<processing_date>\d{8})
+        _(?P<collection_number>\d{2})
+        _(?P<collection_category>[A-Z0-9]{2})
+        _(?P<band>[A-Z0-9_]+)
+        \.
+    """
+
+    tarfile_glob = "*.tar.gz"
```

Comment:
No need to encode this at the class level, it isn't specific to this class. You can just hardcode this where you're planning on using it